be/src/storage/schema_change.cpp

```diff
@@ -262,15 +270,13 @@ bool ChunkMerger::_pop_heap() {
     MergeElement element = _heap.top();
     _heap.pop();
 
-    if (++element.row_index >= element.chunk->num_rows()) {
-        return true;
+    if (++element.row_index < element.chunk->num_rows()) {
+        _heap.push(element);
     }
-
-    _heap.push(element);
     return true;
 }
 
-bool LinkedSchemaChange::process(TabletReader* reader, RowsetWriter* new_rowset_writer, TabletSharedPtr new_tablet,
+bool LinkedSchemaChange::process(ChunkIterator* iter, RowsetWriter* new_rowset_writer, TabletSharedPtr new_tablet,
```

Comment:
Schema change receive TabletReader may be better. The ChunkIterator is transparent to schema change class